src/lib.rs

```diff
@@ -492,15 +601,15 @@ impl Context {
         }
     }
 
-    pub fn get_random(&mut self, sessions: SessionTrio, num_bytes: u16) -> Result<Vec<u8>> {
+    pub fn get_random(&mut self, sessions: Sessions, num_bytes: usize) -> Result<Vec<u8>> {
         let mut buffer = null_mut();
         let ret = unsafe {
             Esys_GetRandom(
                 self.mut_context(),
+                self.session,
                 sessions.0,
                 sessions.1,
-                sessions.2,
-                num_bytes,
+                num_bytes.try_into().unwrap(),
```

Comment:
Maybe a check would be nice that `num_bytes` is less than the maximum unsigned value on 16 bits to avoid possible panicking or just deal with the `try_into()` error as you wish.

Reply:
I was going to raise an issue to look into all uses of `unwrap` and that sort of thing - just getting things to work for now

Reply:
Cool, as long as it is noted somewhere and we do not forget to fix it.